utilities/mng.js

```diff
@@ -0,0 +1,100 @@
+const path = require('path');
+const { MongoClient } = require('mongodb');
+const selectLanguage = require('./selectLanguage');
+const { URL, DBNAME } = require('./dbSettings');
+const textObj = require(path.join(__dirname, '../public/text.json'));
+const connObj = {
+  client: null,
+  db: null,
+  collections: null
+};
+
+/**
+ * connect to the mongodb database using the provided parameters
+ */
+async function connect() {
+  // if we are already connected 
+  // no need to reconnect
+  if (connObj.client && connObj.db) {
+    return;
+  }
+
+  // connect to the local db
+  // client = new MongoClient(URL, { 
+  //   useNewUrlParser: true, 
+  //   useUnifiedTopology: true 
+  // });
+  try {
+    connObj.client = await MongoClient.connect(URL, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    });
+    connObj.db = connObj.client.db(DBNAME);
+  }
+  catch (err) {
+    console.error(err);
+  }
+}
+
+/**
+ * Closes mongodb connection
+ */
+async function close() {
+  (await connObj.client).close();
+}
+
+/**
+ * Return true if the mongodb contains the provided collection
+ * 
+ * @param {string} collName 
+ */
+async function hasCollection(collName) {
+  if (!connObj.collections) {
+    connObj.collections = (await connObj.db.listCollections().toArray())
+      .map(coll => coll['name']);
+  }
+  return connObj.collections.includes(collName);
+}
+
+async function mongoImport(collection, data) {
+  try {
+    const col = await connObj.db.collection(collection);
+    await col.deleteMany();
+    const response = await col.insertOne(data);
+    return true;
+  } catch (error) {
+    console.error(error);
+    return false;
+  }
+}
+
+const importJson = async () => {
+  console.log('Importing data!');
+  await connect();
+  let exists = await hasCollection('intl');
```

Comment:
I think this block of importJson is redundant. The NodeJs MongoDb driver will create a new collection if one does not exist since we're not using strict mode http://mongodb.github.io/node-mongodb-native/3.1/api/Db.html#collection

Reply:
I don't want to rely on the fact that mongodb can do automatic stuff. I think it is more safe to add collections on needs. By the way I don't like the fact that the user can push anything he wants. This could be a security issue since mongodb will insert collections if they don't exist user can spam the server by throwing dummy data.

Reply:
I see what you mean. Though mostly I think this is handled through authentication and form checking. Because of the small scope of the project I wasn't sure about worrying about this issue.